### ISSUE TITLE
Notify about redundant condition in filter_map

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15195]
### Before submitting

- [x] I have searched the existing issues and this hasn't been requested yet.

### Cop name

Style/RedundantConditionInFilterMap

### Is your feature request related to a problem?

This statement (simplified for the masses) probably contained multiple conditions, then it was reduced to one, and the undersigned didn't notice the redundancy (maybe because the line was overflowing and breaking around that area).

Didn't find mentioned before: https://github.com/rubocop/rubocop/issues?q=filter_map%20if

### Describe the solution you'd like

`Condition is redundant when equal to value in filter_map`

### Code examples

```ruby
# bad
items.filter_map { it['foo'] if it['foo'] }

# good
items.filter_map { it['foo'] }
```

### Describe alternatives you've considered

Avoid long lines

**Comment by forthrin** [comment]
@johnbridge314: Will you resolve the PR, or should we deem the cop as too trivial?

**Comment by forthrin** [comment]
Reopen if deemed viable.
